 # <font color=darkblue>Programming for Data Analysis 2022 - Project 2</font>

## <font color=darkblue>Orla Corry</font>
***

## <font color=darkblue>Table of Contents</font>
***

- [Introduction](#Introduction)


- [Importing Libraries](#Importing-Libraries)


- [Reading in the csv file](#Reading-in-the-csv-file)


- [Analysis/review of the dataset](#Analysis/review-of-the-dataset)


- [Literature review on classifiers](#Literature-review-on-classifiers)


- [References](#References)

## <font color=darkblue>Introduction</font>
***

## <font color=darkblue>Importing Libraries</font>
***

For this project I will be importing the following Libraries:

1. **Pandas** - to read in the dataset and to analyse data that is in tabular form

In [2]:
import pandas as pd

In [11]:
%%html
<style>
table{float:left}
</style>

## <font color=darkblue>Reading in the csv file</font>
***
I have sourced the Wesconsin Breast Cancer Dataset from <a href="https://github.com/jeffheaton/aifh/blob/master/vol1/python-examples/datasets/breast-cancer-wisconsin.csv" target="_top">Heaton. J (Github), 2013</a>. To pull the actual csv URL from Heaton's Github, I clicked on the *Raw* tab and copied & pasted the URL that was provided for the raw data. Below, I have read in the the csv file to Python using the built in library <a href="https://pandas.pydata.org/docs/" target="_top">Pandas</a>.


In [13]:
# Reading in the csv file using pandas
# Using dataset from https://github.com/jeffheaton/aifh/blob/master/vol1/python-examples/datasets/breast-cancer-wisconsin.csv
dataset = ("https://raw.githubusercontent.com/jeffheaton/aifh/master/vol1/python-examples/datasets/breast-cancer-wisconsin.csv")
df = pd.read_csv (dataset) #storing the read in csv dataframe as df 

In [16]:
# Printing the dataframe (df) to make sure the csv file read in to Python properly
df
#df.shape

id  clump_thickness  size_uniformity  shape_uniformity  \
0    1000025                5                1                 1   
1    1002945                5                4                 4   
2    1015425                3                1                 1   
3    1016277                6                8                 8   
4    1017023                4                1                 1   
..       ...              ...              ...               ...   
694   776715                3                1                 1   
695   841769                2                1                 1   
696   888820                5               10                10   
697   897471                4                8                 6   
698   897471                4                8                 8   

     marginal_adhesion  epithelial_size bare_nucleoli  bland_chromatin  \
0                    1                2             1                3   
1                    5                7            10                3   
2                    1                2             2                3   
3                    1                3             4                3   
4                    3                2             1                3   
..                 ...              ...           ...              ...   
694                  1                3             2                1   
695                  1                2             1                1   
696                  3                7             3                8   
697                  4                3             4               10   
698                  5                4             5               10   

     normal_nucleoli  mitoses  class  
0                  1        1      2  
1                  2        1      2  
2                  1        1      2  
3                  7        1      2  
4                  1        1      2  
..               ...      ...    ...  
694                1        1      2  
695                1        1      2  
696               10        2      4  
697                6        1      4  
698                4        1      4  

[699 rows x 11 columns]

## <font color=darkblue>Analysis/review of the dataset</font>
***
The Breast Cancer Wisconsin (Original) Data Set is a classification dataset that records measurements under a number of attributes <a href="http://odds.cs.stonybrook.edu/breast-cancer-wisconsin-original-dataset/" target="_top">ODDS. (2022)</a> in order to predict whether a case is either of class benign or class malignant. These attributes will be discussed in more detail later on.

Using the <a href="https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.shape.html" target="_top">df.shape</a> command on the dataframe above, we can see that the dataset comprises of 699 instances/cases (rows) of data across 11 attributes (columns). 

The 699 rows of data were gathered over a period of time which spanned from January 1989 through to November 1991. The data gathered over the 8 periods during this time is organised into 8 groups. 

There are 11 features or attributes to this dataset, all of which are of data type *integer*. Starting at column index 0 through to index 10, the first column contains the *id* of the case/instance which are integers with multiple digits. From column index 1 to index 9 the 9 attributes, namely; *clump thickness*, *size uniformity*, *shape uniformity*, *marginal adhesion*, *epithelial size*, *bare nucleoli*, *bland chromatin*, *normal nucleoli* and *mitoses* are given values ranging from 1 to 10. According to <a href="https://www.journalbinet.com/uploads/2/1/0/0/21005390/67.02.09.2020_analysis_of_wisconsin_breast_cancer_original_dataset_using_data_mining_and_machine_learning_algorithms_for_breast_cancer_prediction.pdf" target="_top">Ahmed, T. et al. (2020)</a> the larger the value for these 9 attruibutes, the greater the chance of the case being malignant. Finally, the last column contains the attribute *class* which contains values either 2 or 4 where 2 represents benign and 4 represents malignant <a href="https://archive.ics.uci.edu/ml/datasets/breast+cancer+wisconsin+(original)" target="_top">Wolberg, W.H. (1992)</a>. **Table 1.** below summarises all of the attributes (excluding the *id** attribute) and what values they can take.

<ins>**Table 1.**</ins>



|*Attributes*      |*Value*|
|:-----------------|:--:
|
|Clump thickness   |1-10|
|Size uniformity   |1-10| 
|Shape uniformity  |1-10|
|Marginal Adheasion|1-10|
|Epithelial size   |1-10| 
|Bare nucleoli     |1-10|
|Bland chromatin   |1-10| 
|Normal nucleoli   |1-10| 
|Mitoses           |1-10|
|Class             |2 = benign & 4= malignant|


## <font color=darkblue>Literature review on classifiers</font>
***
While there are may different variations of definitions of classifiers in machine learning, according to the <a href="https://scikit-learn.org/stable/" target="_top">scikit-learn documentation (2022)</a>, classifiers identify which category an object belongs to. A classifier can be defined as an algorithm that automatically assigns data points to a range of categories or classes and there are two main models; supervised and unsupervised <a href="https://www.indeed.com/career-advice/career-development/classifiers-in-machine-learning" target="_top">Indeed Editorial Team. (2022)</a>.

This section sets out to research some previous studies done using the Wisconsin Breast Cancer (Original) Data Set, and to establish what classifiers were used and compare and contrast their performance. 


A study titled *Study and Analysis of Breast Cancer Data* carried out by <a href="https://www.ijert.org/research/study-and-analysis-of-breast-cancer-data-IJERTCONV5IS21015.pdf" target="_top">Manoli. N.S & Padma. S.K (2017) </a> used two forms of classifiers for their study, namely; the Naive-Bayes Classifier and the support Vector Machine (SVM) classifier.

## <font color=darkblue>References</font>
***

1. Heaton J. Github. breast-cancer-wisconsin.csv (2013) https://github.com/jeffheaton/aifh/blob/master/vol1/python-examples/datasets/breast-cancer-wisconsin.csv


2. Pandas Documentation. (2022). https://pandas.pydata.org/docs/


3. Pandas. pandas.DataFrame.shape. (2022). https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.shape.html


4. ODDS. Breast Cancer Wisconsin (Original) dataset. (2022). http://odds.cs.stonybrook.edu/breast-cancer-wisconsin-original-dataset/


4. Ahmed. T, Omtiaz. N and Karmakar. A.(2020). Analysis of Wisconsin Breast Cancer original dataset using data
mining and machine learning algorithms for breast cancer
prediction
https://www.journalbinet.com/uploads/2/1/0/0/21005390/67.02.09.2020_analysis_of_wisconsin_breast_cancer_original_dataset_using_data_mining_and_machine_learning_algorithms_for_breast_cancer_prediction.pdf


5. Wolberg, W. H. (1992) Breast Cancer Wisconsin (Original) Data Set. https://archive.ics.uci.edu/ml/datasets/breast+cancer+wisconsin+(original)


7. How to align table in Jupyter Notebook. Deepsim Acadamey. (2021). https://www.youtube.com/watch?v=_1jg0FlzpUs


8. scikit-learn. (2022). https://scikit-learn.org/stable/



9. Indeed Editorial Team. Machine Learning Classifiers: Definition and 5 Types. (2022). https://www.indeed.com/career-advice/career-development/classifiers-in-machine-learning


10. Manoli. N.S & Padma. S.K. Study and Analysis of Breast Cancer Data (2017) https://www.ijert.org/research/study-and-analysis-of-breast-cancer-data-IJERTCONV5IS21015.pdf